planner/core/preprocess.go

```diff
@@ -675,6 +675,10 @@ func checkAutoIncrementOp(colDef *ast.ColumnDef, index int) (bool, error) {
 
 func isConstraintKeyTp(constraints []*ast.Constraint, colDef *ast.ColumnDef) bool {
```

Comment:
BTW, can we change the name of this function? I am worrying that somebody will use this function for non-auto-incr columns.

Reply:
It returns whether the column is a constraint key.

I think it's fine to use it for other purpose, at least the change in this PR doesn't break it, because the `CHECK` constraint cannot be regarded as a key (and is not supported by TiDB, as described in the document) :thinking: .